src/FeatureSet.php

```diff
@@ -96,6 +97,11 @@ class FeatureSet
     private $timeGenerator;
 
     /**
+     * @var ValidatorInterface
+     */
+    private $validator;
```

Comment:
Our standard pattern has been to move these things onto `UuidFactoryInterface`. Then, when needing to change the dependencies, we get a factory, set the values there, and then when the methods on `Uuid` are called, they take advantage of that. We should follow the same pattern here.

For example:

``` php
$factory = Uuid::getFactory();
$factory->setValidator(new MyValidator());

if (Uuid::isValid($foo)) {
    // do something
}
```

At this point, `Uuid` is now using your own validator. The internals of `isValid()` would look something like this:

``` php
public static function isValid($uuid)
{
    return self::getFactory()->getValidator()->validate($uuid);
}
```

I think this is what @aztech-dev was suggesting on March 22.